agents/agents-features/agents-features-opentelemetry/src/jvmTest/kotlin/ai/koog/agents/features/opentelemetry/feature/OpenTelemetryTest.kt

```diff
@@ -715,6 +715,172 @@ class OpenTelemetryTest {
+
+            val mockExecutor = getMockExecutor(clock = testClock) {
+                mockLLMAnswer(mockResponse) onRequestEquals userPrompt
+            }
+
+            val agent = createAgent(
+                agentId = agentId,
+                strategy = strategy,
+                promptId = promptId,
+                promptExecutor = mockExecutor,
+                model = model,
+                clock = testClock,
+                temperature = temperature
+            ) {
+                install(OpenTelemetry) {
+                    addSpanExporter(mockExporter)
+                    setVerbose(true)
+                }
+            }
+
+            agent.run(userPrompt)
+
+            val collectedSpans = mockExporter.collectedSpans
+            assertTrue(collectedSpans.isNotEmpty(), "Spans should be created during agent execution")
+
+            agent.close()
+
+            // Print all collected spans for debugging
+            logger.debug { "All collected spans: ${collectedSpans.map { it.name }}" }
+
+            // Check each span
+            // We expect spans for:
+            // 1. Agent creation
+            // 2. Agent run
+            // 3. Start node
+            // 4. Each parallel node (3 nodes)
+            // 5. Merge node
+            // 6. Finish node
+
+            // Verify that we have spans for all parallel nodes
+            val nodeSpanNames = collectedSpans.map { it.name }
+                .filter { it.startsWith("node.") }
+                .sorted()
+            
+            logger.debug { "Node span names: $nodeSpanNames" }
+            
+            // Print all node spans with their attributes for debugging
+            collectedSpans.filter { it.name.startsWith("node.") }.forEach { span ->
+                val attributes = span.attributes.asMap().asSequence().associate { it.key.key to it.value }
+                logger.debug { "Node span: ${span.name}, attributes: $attributes" }
+            }
+            
+            // Check if we have the expected number of node spans (5 nodes)
+            assertEquals(5, nodeSpanNames.size, "Expected 5 node spans but found ${nodeSpanNames.size}")
+            
+            // Check for each specific node span
+            assertTrue(nodeSpanNames.any { it.contains("nodeFirstJoke") }, "First joke node span should be created")
+            assertTrue(nodeSpanNames.any { it.contains("nodeSecondJoke") }, "Second joke node span should be created")
+            assertTrue(nodeSpanNames.any { it.contains("nodeThirdJoke") }, "Third joke node span should be created")
+            assertTrue(nodeSpanNames.any { it.contains("nodeSelectBestJoke") }, "Select best joke node span should be created")
+            assertTrue(nodeSpanNames.any { it.contains("nodeGenerateJokes") }, "Generate jokes node span should be created")
+            
+            // Create expected spans structure
+            val expectedSpans = listOf(
```

Comment:
I would use the fill set of expected spans that we capture in the agent run and compare the actual result with it (if parallel nodes execution does not make the sequence of nodes execution unpredictable). Otherwise, I would, check an exact spans structure for each parallel node, instead of `it.contains(...)` since spans structure should be known here... and would drop the rest of spans assertions (agent invoke, agent create, ... ) as this test target is "nodes".